core/src/main/java/io/questdb/cairo/TableReader.java

```diff
@@ -356,10 +363,24 @@ public int getPartitionIndexByTimestamp(long timestamp) {
         return end / PARTITIONS_SLOT_SIZE;
     }
 
+    public long getPartitionMaxTimestampFromMetadata(int partitionIndex) {
+        long minTimestamp = getPartitionMinTimestampFromMetadata(partitionIndex);
+        // todo: the max timestamp must be within the given partition
+        return txFile.getNextLogicalPartitionTimestamp(minTimestamp) - 1;
```

Comment:
I think this is a wrong implementation. It will not work in the case of split partitions. Logical timestamp will be the next day, but the data boundary is the next partition timestamp in the case of split partitions.